gnmi_subscribe/gnmi_subscribe.go

```diff
@@ -139,24 +142,93 @@ func main() {
 	}
 	switch subscriptionListMode {
 	case pb.SubscriptionList_STREAM:
-		stream()
+		if err := stream(); err != nil {
+			log.Exitf("Error using STREAM mode: %v", err)
+		}
 	case pb.SubscriptionList_POLL:
-		poll()
+		if err := poll(); err != nil {
+			log.Exitf("Error using POLL mode: %v", err)
+		}
 	case pb.SubscriptionList_ONCE:
-		once()
+		if err := once(); err != nil {
+			log.Exitf("Error using ONCE mode: %v", err)
+		}
 	}
 }
 
-func stream() {
-
+func stream() error {
+	for {
+		res, err := subscribeClient.Recv()
+		if err != nil {
+			return err
+		}
+		switch res.Response.(type) {
+		case *pb.SubscribeResponse_SyncResponse:
+			log.Info("SyncResponse received")
+		case *pb.SubscribeResponse_Update:
+			utils.PrintProto(res)
+		default:
+			return errors.New("Unexpected response type")
+		}
+	}
 }
 
-func poll() {
+func poll() error {
+	ready := make(chan bool, 1)
+	ready <- true
+	pollRequest := &pb.SubscribeRequest{Request: &pb.SubscribeRequest_Poll{}}
+	if *updatesOnly {
+		res, err := subscribeClient.Recv()
+		if err != nil {
+			return err
+		}
+		if syncRes := res.GetSyncResponse(); !syncRes {
+			return errors.New("Failed to receive SyncResponse")
```

Comment:
Updates Only flag is set but failed to receive Sync Response first for Poll mode.

Reply:
Sounds good, changed!